### PR TITLE
Calculate feature map shape patchcore

### PR DESCRIPTION
**Pull request by alexriedel1**
Fixing https://github.com/openvinotoolkit/anomalib/issues/40

### CODE REVIEW

File: anomalib/models/patchcore/model.py [hunk 1]
```diff
@@ -163,12 +165,13 @@ def forward(self, input_tensor: Tensor) -> Union[torch.Tensor, Tuple[torch.Tenso
             embedding = self.tiler.untile(embedding)
 
         embedding = self.reshape_embedding(embedding)
+        feature_map_shape = embedding.shape[-2:]
```
Comment (djdameln) [inline anomalib/models/patchcore/model.py:168]: The feature map has to be computed before reshaping the embedding for this to work correctly. So this line needs to be placed above line 167.

**Review by djdameln — CHANGES_REQUESTED** [review]
Thanks @alexriedel1! I spotted one small mistake. Other than that the PR looks fine and it's a great addition.

File: anomalib/models/patchcore/model.py (hunk 1 above)
Comment (alexriedel1) [reply anomalib/models/patchcore/model.py:167]: right!

**Review by djdameln — APPROVED** [review]

**Comment by djdameln** [comment]
There seem to be some code quality issues, see logs [here](https://github.com/openvinotoolkit/anomalib/runs/5588186048?check_suite_focus=true). Did you use [pre-commit](https://openvinotoolkit.github.io/anomalib/guides/using_pre_commit.html#pre-commit-hooks)?

**Comment by alexriedel1** [comment]
> There seem to be some code quality issues, see logs [here](https://github.com/openvinotoolkit/anomalib/runs/5588186048?check_suite_focus=true). Did you use [pre-commit](https://openvinotoolkit.github.io/anomalib/guides/using_pre_commit.html#pre-commit-hooks)?

now I did :)

**Comment by samet-akcay** [comment]
@alexriedel1, I'll run the CI now. We could merge as soon as the tests pass

**Comment by alexriedel1** [comment]
> @alexriedel1, I'll run the CI now. We could merge as soon as the tests pass

ahh sorry I didn't commit after the formatting changes were made, gonna fix it

File: anomalib/models/patchcore/model.py [hunk 2]
```diff
@@ -46,18 +46,20 @@ def __init__(
         self.input_size = input_size
         self.sigma = sigma
 
-    def compute_anomaly_map(self, patch_scores: torch.Tensor) -> torch.Tensor:
+    def compute_anomaly_map(self, patch_scores: torch.Tensor, feature_map_shape: torch.Tensor) -> torch.Tensor:
```
Comment (samet-akcay) [inline anomalib/models/patchcore/model.py:49]: @alexriedel1, is `feature_map_shape` of type `torch.Tensor` or `torch.Size`?

Comment (alexriedel1) [reply anomalib/models/patchcore/model.py]: Right its a `torch.Size`

Comment (samet-akcay) [reply anomalib/models/patchcore/model.py]: Thanks for quickly addressing it

**Comment by samet-akcay** [comment]
Checks complain about some formatting issues. If you could run `black -l 120 anomalib` from the main root directory, black should be able to format the file, and the CI would pass.

**Comment by alexriedel1** [comment]
> Checks complain about some formatting issues. If you could run `black -l 120 anomalib` from the main root directory, black should be able to format the file, and the CI would pass. Apologies for this bureaucracy, but this is needed for the maintanence.

Sorry for making this such a mess :D

**Comment by samet-akcay** [comment]
> Sorry for making this such a mess :D

No problem at all! Thanks a lot for the contribution!

**Review by samet-akcay — APPROVED** [review]

**Review by djdameln — APPROVED** [review]

**Pull request merged**